dali/kernels/imgproc/color_manipulation/brightness_contrast_gpu_test.cu

```diff
@@ -0,0 +1,250 @@
+#include <vector>
+#include <tuple>
+#include "dali/kernels/scratch.h"
+#include "dali/kernels/tensor_shape.h"
+#include "dali/kernels/common/copy.h"
+#include "dali/kernels/test/tensor_test_utils.h"
+#include "dali/kernels/test/kernel_test_utils.h"
+#include "dali/kernels/imgproc/color_manipulation/brightness_contrast_gpu.h"
+
+namespace dali {
+namespace kernels {
+namespace brightness_contrast {
+namespace test {
+
+namespace {
+
+constexpr size_t kNdims = 3;
+
+/**
+ * Rounding to nearest even (like GPU does it)
+ */
+template <class In, class Out>
+std::enable_if_t<std::is_integral<Out>::value, Out> custom_round(float val) {
+  return static_cast<Out>(std::nearbyint(val));
+}
+
+
+template <class In, class Out>
+std::enable_if_t<!std::is_integral<Out>::value, Out> custom_round(float val) {
+  return val;
+}
+
+
+}  // namespace
+
+template <class InputOutputTypes>
+class BrightnessContrastGpuTest : public ::testing::Test {
+  using In = typename InputOutputTypes::In;
+  using Out = typename InputOutputTypes::Out;
+
+ public:
+  BrightnessContrastGpuTest() {
+    input_host_.resize(dataset_size());
+  }
+
+
+  void SetUp() final {
+    std::mt19937_64 rng;
+    UniformRandomFill(input_host_, rng, 0., 10.);
+    calc_output(0);
+    CUDA_CALL(cudaMalloc(&input_device_, sizeof(In) * dataset_size()));
+    CUDA_CALL(cudaMemcpy(input_device_, input_host_.data(), input_host_.size() * sizeof(In),
+                         cudaMemcpyDefault));
+    CUDA_CALL(cudaMallocManaged(&output_, dataset_size() * sizeof(Out)));
+    cudaDeviceSynchronize();
+
+    verify_test();
+  }
+
+
+  In *input_device_;
+  Out *output_;
+  std::vector<In> input_host_;
+  std::vector<Out> ref_output_;
+  std::vector<TensorShape<kNdims>> shapes_ = {{2, 4, 3}};
+  std::vector<float> brightness_ = {4};
+  std::vector<float> contrast_ = {3};
+
+
+  void verify_test() {
+    assert(shapes_.size() == brightness_.size());
+    assert(brightness_.size() == contrast_.size());
+    assert(dataset_size() == input_host_.size());
+    assert(dataset_size() == ref_output_.size());
+  }
+
+
+  void calc_output(int idx) {
+    for (auto in : input_host_) {
+      ref_output_.push_back(custom_round<In, Out>(in * contrast_[idx] + brightness_[idx]));
+    }
+  }
+
+
+  size_t dataset_size() {
+    int ret = 0;
+    for (auto sh : shapes_) {
+      ret += volume(sh);
+    }
+    return ret;
+  }
+};
+
+using TestTypes = std::tuple<int8_t, float>; /* Cause the line below takes RIDICULOUSLY long time to compile */ // NOLINT
+// using TestTypes = std::tuple<uint8_t, int8_t, uint16_t, int16_t, int32_t, float>;
+INPUT_OUTPUT_TYPED_TEST_SUITE(BrightnessContrastGpuTest, TestTypes);
+
+namespace {
+
+template <class GtestTypeParam>
+using TheKernel = BrightnessContrastGpu
+        <typename GtestTypeParam::In, typename GtestTypeParam::Out, kNdims>;
+
+}  // namespace
+
+
+TYPED_TEST(BrightnessContrastGpuTest, check_kernel) {
+  check_kernel<TheKernel<TypeParam>>();
+  SUCCEED();
```

Comment:
what is this? A test succeeds if it didn't fail. What does this call do?

Reply:
Just a verbose info, that it's meant to be this way...

Reply:
I'd remove it

Reply:
Maybe `EXPECT_NO_THROW` or `ASSERT_NO_THROW` would help here?

Reply:
Basically it's compile-time test, I'll remove the `SUCCEED`